### ISSUE TITLE
Partial mock results in a fatal error in PHP 8.2.20

### ISSUE THREAD

**Issue by lzref** [phake/phake#323]
Fresh install of Phake v2.0.2 via Composer (commit hash 15d688aa23e6d5db433accb75feea6c0a1cdd1f1), PHP 8.2.20.

```
<?php

require_once("bootstrap.php");

class ClassToReturn
{

}

class ClassToTest
{
    public function methodToTest(): ClassToReturn
    {
        return new ClassToReturn();
    }
}

$partialMock = \Phake::partialMock(ClassToTest::class);
```

Running this results in the following fatal error:
`PHP Fatal error:  Declaration of ClassToTest_PHAKE6671db34e37c9::methodToTest() must be compatible with ClassToTest::methodToTest(): ClassToReturn in /var/c2/code/care2/vendor/phake/phake/src/Phake/ClassGenerator/EvalLoader.php(58) : eval()'d code on line 29`

So I can't get partial mocks to work at all. Am I doing something wrong?

**Comment by adoy** [comment]
Phake 2 is not maintained anymore. Please upgrade to Phake 4.

**Comment by lzref** [comment]
Thank you!

Weird, but I'm getting version 2.0.2 installed when I run `composer.phar require --dev phake/phake`!

```
 ./composer.json has been updated
Running composer update phake/phake
Loading composer repositories with package information
Updating dependencies
Lock file operations: 1 install, 0 updates, 0 removals
  - Locking phake/phake (v2.0.2)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 1 install, 0 updates, 0 removals
  - Downloading phake/phake (v2.0.2)
  - Installing phake/phake (v2.0.2): Extracting archive
1 package suggestions were added by new dependencies, use `composer suggest` to see details.
Generating autoload files
85 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
No security vulnerability advisories found.
Using version ^2.0 for phake/phake
```

**Comment by adoy** [comment]
That's weird. I tried on my end and I got `Using version ^4.5 for phake/phake`

**Comment by lzref** [comment]
Figured out what the reason was:
```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires phake/phake 4.5 -> satisfiable by phake/phake[v4.5.0].
    - phake/phake v4.5.0 requires doctrine/instantiator ^1.4 -> found doctrine/instantiator[1.4.0, 1.4.1, 1.5.0] but the package is fixed to 1.3.1 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.
```

Using Composer's `--with-all-dependencies` option did work
